pkg/arvo/app/link-proxy-hook.hoon

```diff
@@ -0,0 +1,231 @@
+::  link-proxy-hook: make local pages available to foreign ships
+::
+::    this is a "proxy" style hook, relaying foreign subscriptions into local
+::    stores if permission conditions are met.
+::    the patterns herein should one day be generalized into a proxy-hook lib.
+::
+::    this adopts a very primitive view of groups-store as containing only
+::    groups of interesting (rather than uninteresting) ships. it sets the
+::    permission condition to be that ship must be in group matching the path
+::    it's subscribing to.
+::    we check this on-watch, but also subscribe to groups so that we can kick
+::    subscriptions if needed (eg ship removed from group).
+::
+::    we deduplicate incoming subscriptions on the same path, ensuring we have
+::    exactly one local subscription per unique incoming subscription path.
+::    this comes at the cost of assuming that the store's initial response is
+::    whatever's returned by the scry at that path, but perhaps that should
+::    become part of the stores standard anyway.
+::
+/-  *link, group-store
+/+  default-agent, verb
+|%
++$  state-0
+  $:  %0
+      ::TODO  we use this to detect "first sub started" and "last sub left",
+      ::      but can't we use [wex sup]:bowl for that?
```

Comment:
Spider depends on depends on `wex` and `sup` being correct, so those should work fine.